### PR TITLE
Gui: connect to the tabwidget and check if the task tab was changed

### PR DESCRIPTION
**Pull request by apeltauer**
Add a more generic way to switch back to the latest not task tab.

### CODE REVIEW

**Comment by abdullahtahiriyo** [comment]
@apeltauer 

At a first glance the code looks ok to me. 

However, I fail to identify which is the change in the tab index that you are trying to keep track of (that is obviously different from the showDialog/closedDialog preexisting this commit). Could you please post in the forum an example of what are you trying to solve?

That should help me to identify if there are situations where index change signals should be blocked or not and enable me to test the feature.

Sorry for the inconvenience (I am an integrator in practice for a reason).

NOTES TO SELF:
The changes to oldTabIndex are restricted to the scope of Comboview. Before this commit, this was updated  upon showing a dialog and destroying it via showDialog (spread all over FC) and closedDialog (via the abouttobedestroyed signal).

This commit enables additional updates of what should be considered to be the oldTabIndex every time that the current index changes (via signal currentChanged(int)). This is only raised if the index actually changes (not if the same preexisting index is set).

To be tested: If this oldTabIndex update interferes with legacy behaviour.

**Comment by apeltauer** [comment]
@abdullahtahiriyo this change is for the following use case:
Think about a 3rd, 4th, 5th or 6th tab in the comboview. Then it make sens to save the last changed tab index. Right now in the standard build of FC i think there is no workbench that uses a additional tab. From this point of view, the code adoption makes no sense. But if there is a additional tab, then the last tab is saved.

**Comment by wwmayer** [comment]
To clarify things this feature is to restore the tab index *after* a task dialog was opened.

If there are further tabs (for testing purposes you can uncomment the lines to add the ProjectWidget) then it would be possible already now to reactivate any tab after the dialog was closed.
What wouldn't work is to reactivate the last active tab after the task dialog was opened in case the user has switched. 

Example:
Activate Project tab
Open a task dialog
Activate the Model tab
Switch to the task dialog and close it

Current behaviour: it activates the Project tab. Future behaviour: it activates the Model tab

**Comment by wwmayer** [comment]
Merged.

**Comment by abdullahtahiriyo** [comment]
@apeltauer 

Sorry I did not react inmediately. I have been overloaded with work.

@wwmayer 

Thanks. Precisely what I was worried about is whether what the user wants to see is the tab that was active before starting the task or a tab that was activated in the meantime, upon exiting the task. I am happy with any behaviour, though.

What I must admit (shame on me), is that while I understood your example, I still cannot reproduce it. The reason is that I do not have a "project" tab, only "model" and "tasks". I have looked in the menus and I have search the forum. I even found a snapshot with the three of them in FC 0.14:
https://forum.freecadweb.org/viewtopic.php?style=3&t=6419#p51729

with a vague reference that the "project tab" was a non-funcional file browser that was removed:
https://forum.freecadweb.org/viewtopic.php?style=3&t=6419#p51736

Many tutorials seem to refer to "project tab" and "model tab" as the same reality.

So, basically, I have even done a cursory search in the parameters, but I have not found the way to make a "project" tab appear.
